src/main/java/org/radarcns/management/web/rest/SubjectResource.java

```diff
@@ -220,26 +222,54 @@
     @Timed
     public ResponseEntity<List<SubjectDTO>> getAllSubjects(@ApiParam Pageable pageable,
             @RequestParam(value = "projectName", required = false) String projectName,
-            @RequestParam(value = "externalId", required = false) String externalId)
+            @RequestParam(value = "externalId", required = false) String externalId,
+            @RequestParam(value = "withInactiveParticipants", required = false)
+                    Boolean withInactiveParticipants)
             throws NotAuthorizedException {
         checkPermission(getJWT(servletRequest), SUBJECT_READ);
         log.debug("ProjectName {} and external {}", projectName, externalId);
+        // if not specified do not include inactive patients
+        boolean withInactive = withInactiveParticipants != null ? withInactiveParticipants : false;
         if (projectName != null && externalId != null) {
-            Optional<Subject> subject = subjectRepository
-                    .findOneByProjectNameAndExternalId(projectName, externalId);
+            Optional<Subject> subject = Optional.empty();
+            if (!withInactive) {
```

Comment:
Also a minor thing: perhaps you can do a single check at the start of the method to create the correct list of authorities (either with or without `INACTIVE_PARTICIPANT`), and then use that list in the rest of the method, then you can remove the repeated checks on `withInactive`.

Reply:
indeed. That was the idea. :( I missed it